modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/table/partition/ClientPrimaryPartitionsToNodesGetRequest.java

```diff
@@ -0,0 +1,83 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.handler.requests.table.partition;
+
+import static org.apache.ignite.client.handler.requests.table.ClientTableCommon.readTableAsync;
+import static org.apache.ignite.internal.util.CompletableFutures.nullCompletedFuture;
+
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.internal.client.proto.ClientMessagePacker;
+import org.apache.ignite.internal.client.proto.ClientMessageUnpacker;
+import org.apache.ignite.internal.table.partition.HashPartition;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.NetworkAddress;
+import org.apache.ignite.table.manager.IgniteTables;
+import org.apache.ignite.table.partition.Partition;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Client all primary partitions get request.
+ */
+public class ClientPrimaryPartitionsToNodesGetRequest {
+
+    /**
+     * Process the request.
+     *
+     * @param in Unpacker.
+     * @param out Packer.
+     * @param tables Ignite tables.
+     * @return Future which will complete after request process finished.
+     */
+    public static CompletableFuture<Void> process(
+            ClientMessageUnpacker in,
+            ClientMessagePacker out,
+            IgniteTables tables
+    ) {
+        return readTableAsync(in, tables).thenCompose(table -> {
+            if (table == null) {
+                out.packNil();
+                return nullCompletedFuture();
+            } else {
+                return table.partitionManager()
+                        .primaryReplicasAsync()
+                        .thenAccept(partitions -> {
+                            out.packInt(partitions.size());
+                            for (Entry<Partition, ClusterNode> e : partitions.entrySet()) {
+                                HashPartition partition = (HashPartition) e.getKey();
+
+                                out.packInt(partition.partitionId());
+                                packClusterNode(e.getValue(), out);
+                            }
+                        });
+            }
+        });
+    }
+
+    private static void packClusterNode(@Nullable ClusterNode clusterNode, ClientMessagePacker out) {
+        if (clusterNode == null) {
+            out.packNil();
+        } else {
+            out.packString(clusterNode.id());
+            out.packString(clusterNode.name());
+            NetworkAddress address = clusterNode.address();
+            out.packString(address.host());
+            out.packInt(address.port());
```

Comment:
Let's reuse this part with `ClientClusterGetNodesRequest`. Note `out.packInt(4);` there which allows extending the node info in the future while keeping the protocol compatible.